File: problems/2352_equal_row_and_column_pairs/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        int n = grid.size(), count = 0;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++) {
                bool eq = true;
                for (int k = 0; k < n; k++)
                    if (grid[i][k] != grid[k][j]) { eq = false; break; }
                if (eq) count++;
            }
        return count;
    }
};

File: problems/2352_equal_row_and_column_pairs/solution.py
from typing import List


class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        n = len(grid)
        count = 0
        for i in range(n):
            for j in range(n):
                if all(grid[i][k] == grid[k][j] for k in range(n)):
                    count += 1
        return count
